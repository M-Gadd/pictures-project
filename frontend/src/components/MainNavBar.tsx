//

import React, { useState } from "react";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  NavbarText,
  Row,
  Col,
} from "reactstrap";
import api from "../api";

import { NavLink as NLink } from "react-router-dom";

const MainNavBar = (props: any) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  const loggingout = () => {
    api.logout().then(() => (window.location.href = "/"));
    console.log("Loggging out");
  };

  return (
    <NavLink className="d-flex justify-content-center">
      <Col xs={api.isLoggedIn() ? 6 : 12}>
        <div>
          <Navbar
            className="mainNavBBar"
            // fixed="top"
            color="transparent"
            light
            expand="md"
          >
            {/* <NavbarBrand href="/">reactstrap</NavbarBrand> */}
            <NavbarToggler onClick={toggle} />
            <Collapse isOpen={isOpen} navbar>
              {/* <Nav className={api.isLoggedIn() ? "mx-auto" : "mr-auto"} navbar> */}
              <Nav className="mr-auto" navbar>
                {/* <Row className="d-flex nav-jutstified"> */}
                {/* <Col className="" xs={8}> */}
                <NavItem>
                  <NavLink style={{ color: "white" }} tag={NLink} to="/">
                    Home
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink style={{ color: "white" }} tag={NLink} to="/streetart">
                    My uploads
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink style={{ color: "white" }} tag={NLink} to="/users">
                    Users
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink style={{ color: "white" }} tag={NLink} to="/streetarts">
                    {" "}
                    Street Art
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink style={{ color: "white" }} tag={NLink} to="/streetarts">
                    {" "}
                    I am staging
                  </NavLink>
                </NavItem>
              </Nav>
              <Nav className="ml-auto" navbar>
                {/* </Col> */}
                {/* <Col xs={4}> */}
                {!api.isLoggedIn() && (
                  <NavItem>
                    <NavLink style={{ color: "white" }} tag={NLink} to="/signup">
                      Signup
                    </NavLink>
                  </NavItem>
                )}
                {api.isLoggedIn() && (
                  <NavItem>
                    <NavLink
                      tag={NLink}
                      style={{ color: "white" }}
                      to="#"
                      onClick={() => loggingout()}
                    >
                      Logout
                    </NavLink>
                  </NavItem>
                )}
                {!api.isLoggedIn() && (
                  <NavItem>
                    <NavLink style={{ color: "white" }} tag={NLink} to="/login">
                      Login
                    </NavLink>
                  </NavItem>
                )}
              </Nav>
              {/* </Col> */}
              {/* </Row> */}
              {/* <NavItem>
              <NavLink href="https://github.com/reactstrap/reactstrap">GitHub</NavLink>
            </NavItem> */}
              {/* <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                Options
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem>Option 1</DropdownItem>
                <DropdownItem>Option 2</DropdownItem>
                <DropdownItem divider />
                <DropdownItem>Reset</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown> */}

              {/* <NavbarText>Simple Text</NavbarText> */}
            </Collapse>
          </Navbar>
          <style>{`
            .mainNavBBar{
              color: white
            }

            .custom-toggler .navbar-toggler-icon {
             color: white
            }
          `}</style>
        </div>
      </Col>
    </NavLink>
  );
};

export default MainNavBar;

// import React, { useEffect } from "react";
// import api from "../api";

// export interface MainNavBarProps {}

// const MainNavBar: React.SFC<MainNavBarProps> = () => {
//   // useEffect(() => {}, [api.isLoggedIn()]);

//   const loggingout = () => {
//     api.logout().then(() => (window.location.href = "/"));
//   };

//   console.log(api.isLoggedIn());
//   return (
//     <div>
//       <button className="btn-dark" onClick={() => (window.location.href = "/")}>
//         Home
//       </button>
//       <button className="btn-dark" onClick={() => (window.location.href = "/signup")}>
//         Sign up
//       </button>
//       {api.isLoggedIn() ? (
//         <button className="btn-dark" onClick={() => loggingout()}>
//           Log out
//         </button>
//       ) : (
//         <button className="btn-dark" onClick={() => (window.location.href = "/login")}>
//           Login
//         </button>
//       )}

//       <button className="btn-dark" onClick={() => (window.location.href = "/streetart")}>
//         Profile streetArt
//       </button>
//       <button className="btn-dark" onClick={() => (window.location.href = "/users")}>
//         Users
//       </button>
//       <button className="btn-dark" onClick={() => (window.location.href = "/streetarts")}>
//         All streetArt
//       </button>
//     </div>
//   );
// };

// export default MainNavBar;
