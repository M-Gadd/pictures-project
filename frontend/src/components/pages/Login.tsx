import React from "react";
import { useForm } from "react-hook-form";
import api from "../../api";
import { Input, Row, Col, Container } from "reactstrap";

export interface LoginProps {}

const Login: React.SFC<LoginProps> = () => {
  const { register, handleSubmit, errors } = useForm();

  const onSubmit = (data: any) => {
    api.login(data).then(() => (window.location.href = "/"));
  };

  return (
    <Container>
      <Row className="d-flex justify-content-center">
        <Col xs={5}>
          <form className="mt-5" onSubmit={handleSubmit(onSubmit)}>
            <Input
              type="text"
              placeholder="Email"
              name="Email"
              className="= mt-5"
              innerRef={register({ required: true, pattern: /^\S+@\S+$/i })}
            />
            <Input
              type="password"
              // bsSize="lg"
              placeholder="Password"
              name="Password"
              className=" mt-5"
              innerRef={register({ required: true, min: 3, maxLength: 12 })}
            />

            <input
              style={{ borderRadius: "9%" }}
              className="mt-4 p-2"
              type="submit"
              value="Login"
            />
            <style>{`
  .myInput:focus {
    outline: none;
  }
  `}</style>
          </form>
        </Col>
      </Row>
    </Container>
  );
};

export default Login;
