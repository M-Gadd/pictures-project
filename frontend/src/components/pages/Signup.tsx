import React, { useEffect, useState } from "react";
import api from "../../api";
import {
  Button,
  FormGroup,
  CustomInput,
  Form,
  Input,
  Container,
  Row,
  Col,
  FormText,
} from "reactstrap";
import { useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";

export interface SignupProps {}

const Signup: React.SFC<SignupProps> = () => {
  const [file, setFile] = useState() as any;
  const [uploadedFile, setUploadedFile] = useState() as any;
  const { register, handleSubmit, errors } = useForm();

  const history = useHistory();

  const onSubmit = async (data: any) => {
    await api.signup(data).then((res) => {
      if (uploadedFile) {
        submitUserAvatar(res.user.id);
      }
    });
    // history.push("/");
    window.location.href = "/";
  };

  const handleImageChange = (e: any) => {
    e.preventDefault();
    let reader = new FileReader();
    let thefile = e.target.files[0];

    reader.onloadend = () => {
      setFile(thefile);
      setUploadedFile(reader.result);
    };
    reader.readAsDataURL(thefile);
  };

  const submitUserAvatar = (id: string) => {
    // e.preventDefault();
    const formData = new FormData();
    formData.append("file", file);
    api.uploadImageUser(id, formData);
  };

  return (
    <Container>
      <Row className="d-flex justify-content-center">
        <Col xs={5}>
          <form className="mt-5" onSubmit={handleSubmit(onSubmit)}>
            <Input
              type="text"
              placeholder="First name"
              name="FirstName"
              className="mt-3"
              innerRef={register({ required: true, maxLength: 80 })}
            />
            <Input
              type="text"
              placeholder="Last name"
              name="LastName"
              className="mt-3"
              innerRef={register({ required: true, maxLength: 100 })}
            />
            <Input
              type="text"
              placeholder="Email"
              name="Email"
              className="mt-3"
              innerRef={register({
                required: "Required",
                pattern: {
                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                  message: "invalid email address",
                },
              })}
            />
            {errors.email && errors.email.message}
            <Input
              type="password"
              placeholder="Password"
              name="Password"
              className="mt-3"
              innerRef={register({ required: true, min: 6, maxLength: 12 })}
            />

            <CustomInput
              type="file"
              accept="image/*"
              id="exampleCustomFileBrowser"
              className="mt-3"
              onChange={(e) => handleImageChange(e)}
            />
            {/* <Button
              className="style_photo_button"
              color="primary"
              type="submit"
              disabled={uploadedFile == null || file == null}
            >
              Update Photo
            </Button> */}
            <FormText color="muted">Please upload a photo less than 1MBs</FormText>

            <input
              style={{ borderRadius: "9%" }}
              className="mt-3 p-2"
              // disabled={}
              type="submit"
              value="Signup"
            />
          </form>
        </Col>
      </Row>
    </Container>
  );
};

export default Signup;
