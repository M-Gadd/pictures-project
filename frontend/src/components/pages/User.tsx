import React from "react";
import { useUser } from "../../Hooks/users";
import Default from "../../Assets/default.png";
import { FormText, Button, Row, Col } from "reactstrap";
import api from "../../api";
import EditUser from "../EditUser";

export interface UserProps {}

const User: React.SFC<UserProps> = () => {
  const { user, isLoading } = useUser() as any;

  if (!isLoading) {
    console.log("User: ", user);
  }

  return (
    <div>
      <img
        // style={{ width: "90vw", height: "60vh" }}
        className="img_style_post"
        src={user.PictureURL ? user.PictureURL : Default}
        alt="no one"
      />
      <Row className="mt-3">
        <Col>
          <span style={{ color: "white" }}>
            <h6>
              {user.FirstName} {user.LastName}
            </h6>
          </span>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col>
          <span style={{ color: "white" }}>
            <h6>{user.Email}</h6>
          </span>
        </Col>
      </Row>
      <Row className="mt-3">
        <Col>{user.id === api.getLocalStorageUser().id && <EditUser user={user} />}</Col>
      </Row>
      <style>{`
         .img_style_post {
          border-radius: 50%; 
          height: 200px; 
          width: 200px; 
      `}</style>
    </div>
  );
};

export default User;
