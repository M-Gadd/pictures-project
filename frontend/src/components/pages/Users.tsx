import { useUsers } from "../../Hooks/users";
import React from "react";
import Moment from "react-moment";
import { Row, Col, Card, Container, CardBody, CardTitle } from "reactstrap";
import { useHistory } from "react-router-dom";
import Default from "../../Assets/default.png";

export interface UsersProps {}

const Users: React.SFC<UsersProps> = () => {
  const { users, isLoading } = useUsers();
  const history = useHistory();

  if (!isLoading) {
    console.log(users);
  }

  const getOneUser = (id: string) => {
    history.push("/user", { id: id });
  };
  return (
    <Container>
      <Row>
        {users &&
          users.map((user: any) => (
            <Col key={user.id} xs={4}>
              <Card
                onClick={() => getOneUser(user.id)}
                className=" highlightOnHover style-card-main m-2"
              >
                <CardBody className="style-card-body">
                  <CardTitle>
                    <span>
                      <span style={{ float: "left" }} className="mr-2">
                        <img
                          className="img_style_post"
                          src={user.PictureURL ? user.PictureURL : Default}
                          alt="no one"
                        />
                      </span>
                      <span style={{ fontWeight: "bold" }}>{user.Email}</span>
                    </span>
                    <span style={{ float: "right" }}>
                      <Moment fromNow>{user.CreatedAt}</Moment>
                    </span>
                  </CardTitle>
                  {/* <CardTitle>{post.title}</CardTitle>
                  <CardText>{post.content}</CardText> */}
                  {/* <div className="style-fav">
                    <>
                      <Likes postID={post.id} />
                      <Comments postID={post.id} />
                    </>
                    {authID === post.author_id ? (
                      <div className="ml-auto">
                        <span style={{ marginRight: "20px" }}>
                          <EditPost post={post} />
                        </span>
                        <span>
                          <DeletePost postID={post.id} />
                        </span>
                      </div>
                    ) : (
                      ""
                    )}
                  </div> */}
                </CardBody>
              </Card>
            </Col>
          ))}
      </Row>
      <style>{`
        .highlightOnHover:hover,
        .highlightOnHover:focus {
          cursor: pointer;
          transform: translateY(-1px);
          box-shadow: 0 4px 15px rgba(0, 0, 0, 0.15);
          transition: all 0.3s ease-out;
        }

        .img_style_post {
          border-radius: 50%; 
          height: 30px; 
          width: 30px; 
        }
      `}</style>
    </Container>
  );
};

export default Users;
