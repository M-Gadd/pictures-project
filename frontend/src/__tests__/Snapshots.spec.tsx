import React from "react";
import { create } from "react-test-renderer";
import MainNavBar from "../components/MainNavBar";
import { BrowserRouter as Router } from "react-router-dom";

import Login from "../components/pages/Login";
import Signup from "../components/pages/Signup";
import User from "../components/pages/User";
import Users from "../components/pages/Users";

describe("MainNavBar component", () => {
  test("Matches the snapshot", () => {
    const wrapper = create(
      <Router>
        <MainNavBar />
      </Router>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("Login component", () => {
  test("Matches the snapshot", () => {
    const wrapper = create(<Login />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});

describe("Signup component", () => {
  test("Matches the snapshot", () => {
    const wrapper = create(<Signup />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
// describe("Users component", () => {
//   const data = [
//     {
//       id: "312212",
//       FirstName: "string",
//       LastName: "string",
//       PictureURL: "string",
//       CreatedAt: Date.now(),
//     },
//     {
//       id: "dasdada",
//       FirstName: "string",
//       LastName: "string",
//       PictureURL: "string",
//       CreatedAt: Date.now(),
//     },
//     {
//       id: "d232323",
//       FirstName: "string",
//       LastName: "string",
//       PictureURL: "string",
//       CreatedAt: Date.now(),
//     },
//     {
//       id: "4323213",
//       FirstName: "string",
//       LastName: "string",
//       PictureURL: "string",
//       CreatedAt: Date.now(),
//     },
//   ];
//   test("Matches the snapshot", () => {
//     const wrapper = create(<User />);
//     expect(wrapper.toJSON()).toMatchSnapshot();
//   });
// });
