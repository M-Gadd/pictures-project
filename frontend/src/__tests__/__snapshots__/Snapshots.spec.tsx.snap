// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="d-flex justify-content-center row"
  >
    <div
      className="col-5"
    >
      <form
        className="mt-5"
        onSubmit={[Function]}
      >
        <input
          className="= mt-5 form-control"
          name="Email"
          placeholder="Email"
          type="text"
        />
        <input
          className=" mt-5 form-control"
          name="Password"
          placeholder="Password"
          type="password"
        />
        <input
          className="mt-4 p-2"
          style={
            Object {
              "borderRadius": "9%",
            }
          }
          type="submit"
          value="Login"
        />
        <style>
          
  .myInput:focus {
    outline: none;
  }
  
        </style>
      </form>
    </div>
  </div>
</div>
`;

exports[`MainNavBar component Matches the snapshot 1`] = `
<a
  className="d-flex justify-content-center nav-link"
  onClick={[Function]}
>
  <div
    className="col-12"
  >
    <div>
      <nav
        className="mainNavBBar navbar navbar-expand-md navbar-light bg-transparent"
      >
        <button
          aria-label="Toggle navigation"
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="mr-auto navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/streetart"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                My uploads
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/users"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Users
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/streetarts"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                 
                Street Art
              </a>
            </li>
          </ul>
          <ul
            className="ml-auto navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/signup"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Signup
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/login"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <style>
        
            .mainNavBBar{
              color: white
            }

            .custom-toggler .navbar-toggler-icon {
             color: white
            }
          
      </style>
    </div>
  </div>
</a>
`;

exports[`Signup component Matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="d-flex justify-content-center row"
  >
    <div
      className="col-5"
    >
      <form
        className="mt-5"
        onSubmit={[Function]}
      >
        <input
          className="mt-3 form-control"
          name="FirstName"
          placeholder="First name"
          type="text"
        />
        <input
          className="mt-3 form-control"
          name="LastName"
          placeholder="Last name"
          type="text"
        />
        <input
          className="mt-3 form-control"
          name="Email"
          placeholder="Email"
          type="text"
        />
        <input
          className="mt-3 form-control"
          name="Password"
          placeholder="Password"
          type="password"
        />
        <div
          className="mt-3 custom-file"
        >
          <input
            accept="image/*"
            className="custom-file-input"
            id="exampleCustomFileBrowser"
            onChange={[Function]}
            type="file"
          />
          <label
            className="custom-file-label"
            htmlFor="exampleCustomFileBrowser"
          >
            Choose file
          </label>
        </div>
        <small
          className="form-text text-muted"
        >
          Please upload a photo less than 1MBs
        </small>
        <input
          className="mt-3 p-2"
          style={
            Object {
              "borderRadius": "9%",
            }
          }
          type="submit"
          value="Signup"
        />
      </form>
    </div>
  </div>
</div>
`;
